### PR TITLE
Move db migration CI to GitHub Actions

### PR DESCRIPTION
**Pull request by nschonni**
Another piece split off #23608
Not sure if a bigger trigger is required other than the `db/**` or the gems changing.

### CODE REVIEW

**Review by ykzts — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
> Not sure if a bigger trigger is required other than the `db/**` or the gems changing.

We possibly want to run it on any Ruby code change, as our migrations sometimes refer to models (bad practice and one of the reason the migration tests exist in the first place).

**Comment by nschonni** [comment]
OK, I expanded to all the Ruby file changes, or if the `.ruby-version` changes

**Review by ClearlyClaire — APPROVED** [review]
Unless there is a downside I'm not aware of (slowness, rate limit, …) this looks good! It'd have to be marked as required for merges, though. Not sure how that works.

**Comment by nschonni** [comment]
If you want to mark it as a required job, the filter probably needs to be removed and replaced with something like https://github.com/fkirc/skip-duplicate-actions#skip-ignored-paths, since `paths` doesn't currently play well with branch policies

**Comment by ClearlyClaire** [comment]
Hm, ok… I guess we can run the tests unconditionally for now, and use skip-duplicate-actions or similar in the future.

**Comment by nschonni** [comment]
I split the job in 2 to support a generic `test` required for the branch policy that would also make sure any `test` job passes or is cancelled like the javacript one if something like the following is set in the branch policy
image

Alternately I could fold it back down and the the more specific job name could be selected from that dropdown.

File: .github/workflows/test-migrations-one-step.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          paths: '["Gemfile*", ".ruby-version", "**/*.rb", ".github/workflows/test-migrations-one-step.yml"]'
+
+  test:
+    runs-on: ubuntu-latest
+    needs: pre_job
+    if: needs.pre_job.outputs.should_skip != 'true'
+
+    services:
+      postgres:
+        image: postgres:14.5
```
Comment (nschonni) [inline .github/workflows/test-migrations-one-step.yml:28]: I copied the 14.5 from CircleCI, but docker compose is at the higher `14` and 14.7 looks to be the latest point release.

**Comment by ThisIsMissEm** [comment]
You may also want to look at the concurrency settings on GitHub Actions here, if you force-push a branch or push newer commits, the current CI run isn't stopped/abandoned (i.e., you probably only care than the head of the branch passed)

**Pull request merged**